processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java

```diff
@@ -235,6 +238,9 @@ public String apply(@Nullable AggregatorFactory input)
       final TopNQuery query, final MetricManipulationFn fn
   )
   {
+    final ExtractionFn extractionFn = TopNQueryEngine.canApplyExtractionInPost(query)
```

Comment:
The method this is implemented in is another method that shouldn't really exist on this interface.  I believe it should be possible to do the same thing in the `mergeResults()` method instead of here.  Can we move this logic there and eliminate this?